### PR TITLE
Detect internal IPs, expand whitelist, add output_path option for release builds

### PR DESCRIPTION
**Pull request by diracdeltas**
Fix https://github.com/brave/brave-browser/issues/1706

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "pull_l10n": "node ./scripts/commands.js pull_l10n",
     "chromium_rebase_l10n": "node ./scripts/commands.js chromium_rebase_l10n",
     "test": "node ./scripts/commands.js test",
-    "test-security": "npm audit && npm run network-audit"
+    "test-security": "npm audit && node ./scripts/commands.js start --enable_brave_update --network_log --user_data_dir_name=brave-network-test"
```
Comment (diracdeltas) [inline package.json]: @mihaiplesa i updated this so now `npm run test-security -- --output_path="/path/to/brave/binary"` should work

**Comment by mihaiplesa** [comment]
which branches will this be uplifted to? I'd suggest 0.55.x and up

**Comment by diracdeltas** [comment]
0.55 and up seems ok to me. @bbondy ?

**Comment by bbondy** [comment]
master: dc482e09800edab4136fee6e5705f547dfd1443c
0.57.x: 6ea462f99c6c38a97609371be27aa1e7b67898b8
0.56.x: bb4cb935c8bf675ba9fb1f6a2c8aa4e6c4beffaa
0.55.x: 20c0c7af6eec96835faa4cd588dec68a8b8a15e1

**Pull request merged**